modules/engine/src/main/scala/observe/engine/Engine.scala

```diff
@@ -460,13 +460,15 @@ class Engine[F[_]: MonadError[*[_], Throwable]: Logger, S, U](stateL: Engine.Sta
     initialState: S,
     f:            (A, S) => F[(S, B, Option[Stream[F, A]])]
   )(implicit ev:  Concurrent[F]): Stream[F, B] =
-    Stream.eval(fs2.concurrent.Queue.unbounded[F, Stream[F, A]]).flatMap { q =>
-      Stream.eval_(q.enqueue1(input)) ++
-        q.dequeue.parJoinUnbounded
+    Stream.eval(cats.effect.std.Queue.unbounded[F, Stream[F, A]]).flatMap { q =>
+      Stream.exec(q.offer(input)) ++
```

Comment:
Hadn't seen `exec` before, what does it do?

Reply:
It executes an F[Unit] for its side effect only. In this case, to add an element in the `Queue`